src/files/getFilesByProgramId.ts

```diff
@@ -0,0 +1,52 @@
+import { EgoAccessToken, EgoJwtManager } from "auth";
+import { FILES_SERVICE_URL, FILES_STREAM_SIZE } from "config";
+import logger from "logger";
+import fetch from "node-fetch";
+import promiseRetry from "promise-retry";
+import { File } from "./types";
+
+export const getFilesByProgramId = async (
+  egoJwtManager: EgoJwtManager,
+  programId: string,
+  page: number
+): Promise<File[]> => {
+  const jwt = (await egoJwtManager.getLatestJwt()) as EgoAccessToken;
+  const accessToken = jwt.access_token;
+  const url = `${FILES_SERVICE_URL}/files?page=${page}&limit=${FILES_STREAM_SIZE}&programId=${programId}`;
```

Comment:
this works, but you may find it easier to use some tools to handle URL construction. the JS URL API and `url-join` library that is already in this repo.

```js
const fileQueryUrl = new URL(urljoin(FILES_SERVICE_URL, 'files'));

fileQueryUrl.searchParams.append("page", page);
fileQueryUrl.searchParams.append("limit", FILES_STREAM_SIZE);
fileQueryUrl.searchParams.append("programId", programId);
```